lldb/source/API/SBProcess.cpp

```diff
@@ -810,6 +809,112 @@ const char *SBProcess::GetBroadcasterClass() {
   return ConstString(Process::GetStaticBroadcasterClass()).AsCString();
 }
 
+lldb::SBAddressRangeList
+SBProcess::FindRangesInMemory(const void *buf, uint64_t size,
+                              SBAddressRangeList &ranges, uint32_t alignment,
+                              uint32_t max_matches) {
```

Comment:
I didn't realize how many error conditions that can go wrong here, and logging isn't the place to do that, so we should probably add a `SBError &error` argument to the end here and report the error back to the user:
```
lldb::SBAddressRangeList
SBProcess::FindRangesInMemory(const void *buf, uint64_t size,
                              SBAddressRangeList &ranges, uint32_t alignment,
                              uint32_t max_matches, SBError &error) {
```